component/src/test/java/org/wso2/extension/siddhi/map/text/sourcemapper/util/HttpTestUtil.java

```diff
@@ -0,0 +1,170 @@
+                                .getBytes()));
+            }
+            HttpServerUtil.setHeader(urlConn, "Content-Type", mapping);
+            HttpServerUtil.setHeader(urlConn, "HTTP_METHOD", "POST");
+            HttpServerUtil.writeContent(urlConn, event);
+            assert urlConn != null;
+            logger.info("Event response code " + urlConn.getResponseCode());
+            logger.info("Event response message " + urlConn.getResponseMessage());
+            urlConn.disconnect();
+        } catch (IOException e) {
+            HttpServerUtil.handleException("IOException occurred while running the HttpsSourceTestCaseForSSL", e);
+        }
+    }
+
+    public void httpsPublishEvent(String event, String baseURI, Boolean auth, String mapping)
+            throws KeyManagementException {
+        try {
+            System.setProperty("javax.net.ssl.trustStore", System.getProperty("carbon.home") + "/resources/security/" +
+                    "client-truststore.jks");
+            System.setProperty("javax.net.ssl.trustStorePassword", "wso2carbon");
+            Security.addProvider(new com.sun.net.ssl.internal.ssl.Provider());
+            char[] passphrase = "wso2carbon".toCharArray(); //password
+            KeyStore keystore = KeyStore.getInstance("JKS");
+            keystore.load(new FileInputStream(System.getProperty("carbon.home") + "/resources/security/" +
+                    "client-truststore.jks"), passphrase); //path
+            TrustManagerFactory tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            tmf.init(keystore);
+            SSLContext context = SSLContext.getInstance("TLS");
+            TrustManager[] trustManagers = tmf.getTrustManagers();
+            context.init(null, trustManagers, null);
+            SSLSocketFactory sf = context.getSocketFactory();
+            URL url = new URL(baseURI);
+            HttpsURLConnection httpsCon = (HttpsURLConnection) url.openConnection();
+            httpsCon.setSSLSocketFactory(sf);
+            httpsCon.setRequestMethod("POST");
+            httpsCon.setRequestProperty("Content-Type", mapping);
+            httpsCon.setRequestProperty("HTTP_METHOD", "POST");
+            if (auth) {
+                httpsCon.setRequestProperty("Authorization",
+                        "Basic " + java.util.Base64.getEncoder().encodeToString(("admin" + ":" + "admin").getBytes()));
+            }
+            httpsCon.setDoOutput(true);
+            OutputStreamWriter out = new OutputStreamWriter(httpsCon.getOutputStream());
+            out.write(event);
+            out.close();
+            logger.info("Event response code " + httpsCon.getResponseCode());
+            logger.info("Event response message " + httpsCon.getResponseMessage());
+            httpsCon.disconnect();
+        } catch (NoSuchAlgorithmException | CertificateException | KeyStoreException | IOException e) {
+            logger.error(e);
```

Comment:
Would be better if these exceptions are handled individually, so a specific log can be printed.